src/WebKit/Enumerators.cs

```diff
@@ -24,23 +26,25 @@ internal class IndexedContainerEnumerator<T> : IEnumerator<T> {
 
 		public T Current {
 			get {
-				return _container [_index];
+				return _container! [_index];
 			}
 		}
 
-		object IEnumerator.Current {
+		object? IEnumerator.Current {
 			get { return ((IEnumerator<T>) this).Current; }
 		}
 
 		public bool MoveNext () {
+			if (_container is null)
+				return false;
```

Comment:
This is not correct. MoveNext can throw and throws and exception, please read https://docs.microsoft.com/en-us/dotnet/api/system.collections.ienumerator.movenext?view=net-6.0

Returning false has a meaning: 

> true if the enumerator was successfully advanced to the next element; false if the enumerator has passed the end of the collection.

That means that you are telling the user of the API that everything was ok and that you reached the end, that is not the case. We should throw an InvalidOperationException.

Reply:
`_container` can only be null if the object has been disposed, so an `ObjectDisposedException` sounds natural